x-pack/metricbeat/module/iis/application_pool/reader.go

```diff
@@ -263,7 +263,7 @@ func getw3wpProceses() (map[int]string, error) {
 func getProcessIds(counterValues map[string][]pdh.CounterValue) []WorkerProcess {
 	var workers []WorkerProcess
 	for key, values := range counterValues {
-		if strings.Contains(key, "\\ID Process") {
+		if strings.Contains(key, "\\ID Process") && values[0].Measurement != nil {
```

Comment:
Just curious, do we need to check `values` length first? like `len(values) > 0`

Reply:
a counter value is added by default in pdh_query_windows.go, if it does not contain the Measurement it will contain more info on the error. This is also picked up later.